astropy/utils/data.py

```diff
@@ -232,6 +233,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         try:
             import bz2
         except ImportError:
+            fileobj.close()
```

Comment:
You mean it is not handled by the
```
    finally:
        for fd in close_fds:
            fd.close()
        for fd in delete_fds:
            os.remove(fd.name)
```
if the `ValueError` is raised? But that was exactly the existing situation for the bzip2 case; again I suspect the interpreter without bzip2 support is such a rare case that at least it is not frequently tested. At least when I tried to emulate this on the master branch (forcing the import to fail) I got two `InconsistentTableError`s from `test_read`. Then the `fd_close()` should be added in both cases,right?